src/virtualenv/discovery/windows/__init__.py

```diff
@@ -19,13 +26,17 @@ def propose_interpreters(spec, cache_dir, env):
     )
 
     for name, major, minor, arch, exe, _ in existing:
-        # pre-filter
-        if name in ("PythonCore", "ContinuumAnalytics"):
-            name = "CPython"
-        registry_spec = PythonSpec(None, name, major, minor, None, arch, exe)
-        if registry_spec.satisfies(spec):
+        # Map well-known/most common organizations to a Python implementation, use the org name as a fallback for
+        # backwards compatibility.
+        implementation = _IMPLEMENTATION_BY_ORG.get(name, name)
+
+        # Pre-filtering based on Windows Registry metadata, for CPython only
+        skip_pre_filter = implementation != "CPython"
+        registry_spec = PythonSpec(None, implementation, major, minor, None, arch, exe)
+        if skip_pre_filter or registry_spec.satisfies(spec):
```

Comment:
Alternatively, we add a `impl_must_match` argument to this method and set it to `False` in this case instead of skipping pre-filtering for non-CPython. But that would be a bigger change.